src/request/sign-polygon-transaction/SignPolygonTransactionApi.js

```diff
@@ -127,25 +110,57 @@ class SignPolygonTransactionApi extends PolygonRequestParserMixin(TopLevelApi) {
             if (!['refund'].includes(description.name)) {
                 throw new Errors.InvalidRequestError('Requested Polygon contract method is invalid');
             }
+        } else if (forwardRequest.to === CONFIG.USDC_SWAP_CONTRACT_ADDRESS) {
+            const usdcTransferContract = new ethers.Contract(
+                CONFIG.USDC_SWAP_CONTRACT_ADDRESS,
+                PolygonContractABIs.SWAP_CONTRACT_ABI,
+            );
+
+            /** @type {PolygonSwapDescription | PolygonSwapWithApprovalDescription} */
+            description = (usdcTransferContract.interface.parseTransaction({
+                data: forwardRequest.data,
+                value: forwardRequest.value,
+            }));
+
+            if (description.args.token !== CONFIG.BRIDGED_USDC_CONTRACT_ADDRESS) {
+                throw new Errors.InvalidRequestError('Invalid USDC token contract in request data');
+            }
+
+            if (!['swap', 'swapWithApproval'].includes(description.name)) {
+                throw new Errors.InvalidRequestError('Requested Polygon contract method is invalid');
+            }
+
+            // Ensure swap `targetAmount` is not too low
+            const inputAmount = /** @type {PolygonSwapDescription | PolygonSwapWithApprovalDescription} */ (description)
+                .args
+                .amount;
+            const targetAmount = /** @type {PolygonSwapDescription | PolygonSwapWithApprovalDescription} */ (description) // eslint-disable-line max-len
+                .args
+                .targetAmount;
```

Comment:
No need to cast here as `description` already has the desired type.
```suggestion
            const { inputAmount, targetAmount } = description.args;
```

Reply:
Does it for you? In my editor, it does not (the type annotation in line 119 doesn't actually work to narrow `description`).

Reply:
What I could do is 
```js
const { inputAmount, targetAmount } = /** @type {PolygonSwapDescription | PolygonSwapWithApprovalDescription} */ (description).args;
```

Reply:
`yarn typecheck` has no complaints.

Reply:
Yes, because the variables are happily assigned `any`. Which works, but is not helpful.

Reply:
Yeah I have the same issue as Sisou, `parseTransaction` seems to return a `TransactionDescription` which is apparently not assignable to `description`'s type, thus giving me an error line 120.